src/main/java/bio/terra/service/filedata/DrsService.java

```diff
@@ -199,21 +218,70 @@ public DRSAccessURL getAccessUrlForObjectId(
 
     DrsId drsId = drsIdService.fromObjectId(objectId);
     Snapshot snapshot = snapshotService.retrieve(UUID.fromString(drsId.getSnapshotId()));
-    GoogleProjectResource projectResource = snapshot.getProjectResource();
 
-    DRSAccessMethod matchingAccessMethod =
-        getAccessMethodMatchingAccessId(accessId, drsObject)
-            .orElseThrow(
-                () ->
-                    new IllegalArgumentException(
-                        "No matching access ID " + accessId + " was found on object " + objectId));
+    BillingProfileModel billingProfileModel =
+        snapshot
+            .getFirstSnapshotSource()
+            .getDataset()
+            .getDatasetSummary()
+            .getDefaultBillingProfile();
+
+    Supplier<IllegalArgumentException> illegalArgumentExceptionSupplier =
+        () -> {
+          throw new IllegalArgumentException("No matching access ID was found for object");
+        };
+
+    CloudPlatformWrapper wrapper = CloudPlatformWrapper.of(billingProfileModel.getCloudPlatform());
+    if (wrapper.isGcp()) {
+      DRSAccessMethod matchingAccessMethod =
+          getAccessMethodMatchingAccessId(accessId, drsObject, TypeEnum.GS)
+              .orElseThrow(illegalArgumentExceptionSupplier);
```

Comment:
I think stylistically, we've always just done this inline instead of defining an explicit supplier variable